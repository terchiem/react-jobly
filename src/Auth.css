.Auth {
  width: 40vw;
  margin: 60px auto;
  padding: 10px 20px;
  background-color: #fff;
  border: 1px solid #ccc;
  box-shadow: 1px 1px 2px 0 rgba(0, 0, 0, 0.2);
  position: relative;
}

.Auth-toggle {
  position: absolute;
  top: -36px;
  right: 0;
  margin: 4px;
}

.Auth-toggle button {
  padding: 6px 10px;
  font-size: 0.8em;
}

.Auth-active {
  background-color: #ccc;
}