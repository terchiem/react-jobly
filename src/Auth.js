import React, { useState } from 'react';
import './Auth.css';

// components
import LoginForm from './LoginForm';
import SignUpForm from './SignUpForm';

/** Displays a form to log in or sign up */

function Auth() {

  const [loginForm, toggleLoginForm] = useState(true);

  // TODO: conditional styling for current active form

  return (
    <div className="Auth">
      <div className="Auth-toggle">
        <button onClick={() => toggleLoginForm(true)}>Login</button>
        <button onClick={() => toggleLoginForm(false)}>Sign up</button>
      </div>

      { loginForm ? <LoginForm /> : <SignUpForm /> }
    </div>
  );
}

export default Auth;