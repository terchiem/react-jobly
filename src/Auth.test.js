import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Auth from './Auth';

test('renders without crashing', () => {
  render(
    <MemoryRouter>
      <Auth />
    </MemoryRouter>
  );
});

test('matches snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Auth />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});