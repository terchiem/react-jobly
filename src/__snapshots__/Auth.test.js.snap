// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Auth"
  >
    <div
      class="Auth-toggle"
    >
      <button
        class="Auth-active"
      >
        Login
      </button>
      <button
        class="Auth-button"
      >
        Sign up
      </button>
    </div>
    <form
      class="Form"
    >
      <div
        class="Form-group"
      >
        <label
          class="Form-label"
          for="username"
        >
          Username
        </label>
        <input
          class="Form-input"
          id="username"
          name="username"
          value=""
        />
      </div>
      <div
        class="Form-group"
      >
        <label
          class="Form-label"
          for="password"
        >
          Password
        </label>
        <input
          class="Form-input"
          id="password"
          name="password"
          type="password"
          value=""
        />
      </div>
      <button
        class="Form-submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
